### PR TITLE
Removing unnecessary code, removing alt, actions and store, and makin…

### PR DESCRIPTION
**Pull request by EdwinGuzman**
…g the mobile header state-based.

Removed the Actions and the Header Store and made the mobile header be state-based instead. Currently up on [Staff Pick](staff-picks-dev.us-east-1.elasticbeanstalk.com/books-music-dvds/recommendations/best-books/ya). Also removed components that are not currently being used.

I did NOT remove `dgx-alt-center` because I don't know if it's actually needed, unless if `dgx-feature-flags` brings in its own `alt` package.

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Updated this PR and removed `dgx-alt-center`.

File: src/components/Header/Header.js [hunk 1]
```diff
@@ -125,26 +122,20 @@ class Header extends React.Component {
   }
 
   componentWillUnmount() {
-    HeaderStore.unlisten(this.onChange.bind(this));
     // Listen on FeatureFlags Store updates
     FeatureFlags.store.unlisten(this.onFeatureFlagsChange.bind(this));
     // Removing event listener to minimize garbage collection
   }
 
   onChange() {
```
Comment (nonword) [inline src/components/Header/Header.js]: Is this a core react hook? Don't see it documented outside of html5 interactive elements. Is it fired when the component change?

File: src/components/Header/Header.js [hunk 2]
```diff
@@ -125,26 +122,20 @@ class Header extends React.Component {
...
-      )
-    );
+    this.setState({
+      loginCookieValue: this.state.loginCookieValue,
+      patronName: this.state.patronName,
+      patronInitial: this.state.patronInitial,
+      patronDataReceived: this.state.patronDataReceived,
+      isFeatureFlagsActivated: {},
+      logOutUrl: this.state.logOutUrl,
+    });
```
Comment (nonword) [inline src/components/Header/Header.js]: What is achieved by setting a state variable to the value of that variable in state? I think `setState` only sets those values that you pass in, so I don't think there's a need to pass every state variable in if you're not chanigng it. This seems like it should have no effect? Does this trigger something nonobvious?

File: src/components/Header/Header.js (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Header/Header.js]: It is a core react hook - https://reactjs.org/docs/react-component.html#componentwillunmount

It gets fired when a component gets removed from the DOM.

Comment (nonword) [reply src/components/Header/Header.js]: Sorry I meant `onChange`

File: src/components/Header/Header.js (hunk 2 above)
Comment (EdwinGuzman) [reply src/components/Header/Header.js]: Right, in this case, it's not needed here. Not sure how it ended up here but will remove it.

File: src/components/Header/Header.js (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Header/Header.js]: Ah, it's not. That function can be named anything, but `onChange` is a convention. It's the function that's used to listen and not listen to the Store. http://alt.js.org/guide/view/

Comment (nonword) [reply src/components/Header/Header.js]: I see! Thanks. Looks like, with the removal of the store, `onChange` is no longer called, so the whole method can be deleted https://github.com/NYPL/dgx-header-component/blob/44df90663cb0df9d0ce5e5cfbe18bf7cdc72f98c/src/components/Header/Header.js#L130-L138

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -125,26 +122,19 @@ class Header extends React.Component {
...
-          patronName: this.state.patronName,
-          patronInitial: this.state.patronInitial,
-          patronDataReceived: this.state.patronDataReceived,
-          isFeatureFlagsActivated: {},
-          logOutUrl: this.state.logOutUrl,
-        },
-        HeaderStore.getState()
-      )
-    );
+    this.setState({
```
Comment (rhernand3z) [inline src/components/Header/Header.js]: do you even need an onChange method? we aren't expecting anything to trigger it. The login state objects might but setState() will take care of that right?

Comment (EdwinGuzman) [reply src/components/Header/Header.js]: You're right. Removing `onChange`. Thanks @nonword  and @rhernand3z

**Comment by EdwinGuzman** [comment]
Yes, the click function will trigger updating the state.

File: src/components/Header/Header.js (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Header/Header.js]: Correct, I did delete it just recently.

**Pull request merged**